homeassistant/components/icloud/device_tracker.py

```diff
@@ -1,491 +1,62 @@
-import homeassistant.util.dt as dt_util
-from homeassistant.util.location import distance
-from homeassistant.util.async_ import run_callback_threadsafe
+from . import DATA_ICLOUD, SIGNAL_UPDATE_ICLOUD, IcloudDevice
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_ACCOUNTNAME = 'account_name'
-CONF_MAX_INTERVAL = 'max_interval'
-CONF_GPS_ACCURACY_THRESHOLD = 'gps_accuracy_threshold'
 
-# entity attributes
-ATTR_ACCOUNTNAME = 'account_name'
-ATTR_INTERVAL = 'interval'
-ATTR_DEVICENAME = 'device_name'
-ATTR_BATTERY = 'battery'
-ATTR_DISTANCE = 'distance'
-ATTR_DEVICESTATUS = 'device_status'
-ATTR_LOWPOWERMODE = 'low_power_mode'
-ATTR_BATTERYSTATUS = 'battery_status'
-
-ICLOUDTRACKERS = {}
-
-_CONFIGURING = {}
-
-DEVICESTATUSSET = ['features', 'maxMsgChar', 'darkWake', 'fmlyShare',
-                   'deviceStatus', 'remoteLock', 'activationLocked',
-                   'deviceClass', 'id', 'deviceModel', 'rawDeviceModel',
-                   'passcodeLength', 'canWipeAfterLock', 'trackingInfo',
-                   'location', 'msg', 'batteryLevel', 'remoteWipe',
-                   'thisDevice', 'snd', 'prsId', 'wipeInProgress',
-                   'lowPowerMode', 'lostModeEnabled', 'isLocating',
-                   'lostModeCapable', 'mesg', 'name', 'batteryStatus',
-                   'lockedTimestamp', 'lostTimestamp', 'locationCapable',
-                   'deviceDisplayName', 'lostDevice', 'deviceColor',
-                   'wipedTimestamp', 'modelDisplayName', 'locationEnabled',
-                   'isMac', 'locFoundEnabled']
-
-DEVICESTATUSCODES = {
-    '200': 'online',
-    '201': 'offline',
-    '203': 'pending',
-    '204': 'unregistered',
-}
-
-SERVICE_SCHEMA = vol.Schema({
-    vol.Optional(ATTR_ACCOUNTNAME): vol.All(cv.ensure_list, [cv.slugify]),
-    vol.Optional(ATTR_DEVICENAME): cv.slugify,
-    vol.Optional(ATTR_INTERVAL): cv.positive_int
-})
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_USERNAME): cv.string,
-    vol.Required(CONF_PASSWORD): cv.string,
-    vol.Optional(ATTR_ACCOUNTNAME): cv.slugify,
-    vol.Optional(CONF_MAX_INTERVAL, default=30): cv.positive_int,
-    vol.Optional(CONF_GPS_ACCURACY_THRESHOLD, default=1000): cv.positive_int
-})
-
-
-def setup_scanner(hass, config: dict, see, discovery_info=None):
-    """Set up the iCloud Scanner."""
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    account = config.get(CONF_ACCOUNTNAME, slugify(username.partition('@')[0]))
-    max_interval = config.get(CONF_MAX_INTERVAL)
-    gps_accuracy_threshold = config.get(CONF_GPS_ACCURACY_THRESHOLD)
-
-    icloudaccount = Icloud(hass, username, password, account, max_interval,
-                           gps_accuracy_threshold, see)
-
-    if icloudaccount.api is not None:
-        ICLOUDTRACKERS[account] = icloudaccount
-
-    else:
-        _LOGGER.error("No ICLOUDTRACKERS added")
+def setup_scanner(hass, config, see, discovery_info=None):
```

Comment:
After moving this integration to use config entry, the device tracker platform should subclass the new `TrackerEntity` class in the device tracker component.